checkov/github_actions/runner.py

```diff
@@ -102,3 +117,34 @@ def get_images(self, file_path: str) -> set[Image]:
                     images.add(image_obj)
 
         return images
+
+    @staticmethod
+    def get_start_and_end_lines(key: str) -> list[int]:
+        check_name = key.split('.')[-1]
+        try:
+            start_end_line_bracket_index = check_name.index('[')
+        except ValueError:
+            return [-1, -1]
+        return [int(x) for x in check_name[start_end_line_bracket_index + 1: len(check_name) - 1].split(':')]
+
+    @staticmethod
+    def resolve_job_name(definition: dict[str, Any], start_line: int, end_line: int) -> str:
+        for key, job in definition.get('jobs', {}).items():
+            if key in [START_LINE, END_LINE]:
+                continue
+            if job[START_LINE] <= start_line <= end_line <= job[END_LINE]:
+                return str(key)
+        return ""
+
+    @staticmethod
+    def resolve_step_name(job_definition: dict[str, Any], start_line: int, end_line: int) -> str:
+        for step in job_definition.get('steps', {}):
+            if step[START_LINE] <= start_line <= end_line <= step[END_LINE]:
+                try:
+                    name = step["name"]
+                except KeyError:
+                    name = step[list(step.keys())[0]]
```

Comment:
```suggestion
                    name = step[next(iter(step.keys()))]
```
then you don't have to iterate through everything first to just grab the first entry